server/plugin/mm_34646_token_refresh.go

```diff
@@ -0,0 +1,125 @@
+package plugin
+
+import (
+	"context"
+	"encoding/json"
+	"net/http"
+	"time"
+
+	"github.com/google/go-github/v31/github"
+	"github.com/pkg/errors"
+)
+
+const pageSize = 100
+const delayBetweenPages = 5 * time.Minute
+
+func (p *Plugin) forceResetAllMM34646() error {
+	config := p.getConfiguration()
+	ctx := context.Background()
+
+	for page := 0; ; page++ {
+		keys, appErr := p.API.KVList(page, pageSize)
+		if appErr != nil {
+			return appErr
+		}
+
+		for _, key := range keys {
+			data, appErr := p.API.KVGet(key)
+			if appErr != nil {
+				p.API.LogWarn("failed to inspect key", "key", key, "error",
+					appErr.Error())
+				continue
+			}
+			tryInfo := GitHubUserInfo{}
+			err := json.Unmarshal(data, &tryInfo)
+			if err != nil {
+				// too noisy to report
+				continue
+			}
+			if tryInfo.Token == nil || tryInfo.Token.AccessToken == "" {
+				// too noisy to report
+				continue
+			}
+
+			info, errResp := p.getGitHubUserInfo(tryInfo.UserID)
+			if errResp != nil {
+				p.API.LogError("failed to retrieve GitHubUserInfo", "key", key, "user_id", tryInfo.UserID,
+					"error", errResp.Error())
+				continue
+			}
+
+			_, err = p.forceResetUserTokenMM34646(ctx, config, *info)
+			if err != nil {
+				p.API.LogError("failed to reset GitHub user token", "key", key, "user_id", tryInfo.UserID,
+					"error", err.Error())
+				continue
+			}
+		}
+
+		if len(keys) < pageSize {
+			break
+		}
+		time.Sleep(delayBetweenPages)
+	}
+	return nil
+}
+
+func (p *Plugin) forceResetUserTokenMM34646(ctx context.Context, config *Configuration, info GitHubUserInfo) (string, error) {
+	if info.MM34646ResetTokenDone {
+		return info.Token.AccessToken, nil
+	}
+
+	client, err := p.getResetUserTokenMM34646Client(config)
+	if err != nil {
+		p.API.LogInfo("Failed to create a special GitHub client to refresh the user's token", "error", err.Error())
+	}
+
+	req, apiErr := client.NewRequest(http.MethodPatch,
```

Comment:
Can the API method [`Reset`](https://pkg.go.dev/github.com/google/go-github/v38/github#AuthorizationsService.Reset) be used here instead of a custom one?

Reply:
I couldn't find the basic auth APIs/cli, thx for the pointer (I was only searching for `ResetToken`)